### ISSUE TITLE
Bug: Cards wrap and overlay in Player Hand display beyond 2nd card

### ISSUE THREAD

**Issue by riverwanderer** [vassalengine/vassal#10819]
Demo includes full instructions once New Game is started (to use, rename zip to .vmod):
[Hand bug demo.zip](https://github.com/vassalengine/vassal/files/7599679/Hand.bug.demo.zip)

Forum thread: https://forum.vassalengine.org/t/bug-player-hand-piece-may-snap-to-stack-and-not-show-in-the-row/73087

It is worth fixing this bug because it detracts from the the convenience of the Player Hand. The alternative incurs the overhead of a Trigger and Send to Location trait combo on each card.

**Comment by riverwanderer** [comment]
I have tested this under Vassal v3.2.17 and the bug exists in that version also.

Testing also with an ordinary Private Window, as might be expected, the pieces (cards) all snap and stack to the expected point. Indeed, even in a Player Hand all pieces are at the expected point. However, in the Player Hand cards do not always join the original stack.

By adding a SumStack diagnostic text label onto the cards, I can see that the first card that reaches the home position by "Drag+Drop+Snap to Grid" does not in fact join the stack. It forms a new stack super-imposed on the one that is already there:
![image](https://user-images.githubusercontent.com/58135975/143677982-fc41e158-f9cb-4195-a0e1-7908522aa6ba.png)

Subsequent drag and drop to an empty space will form a stack with the first drag & dropped card; repeat to establish another super-imposed stack of 2 cards etc. 

[Hand bug demo.zip](https://github.com/vassalengine/vassal/files/7611595/Hand.bug.demo.zip)
 is updated to include diagnostics and ability to Send to Location to several cards right of the home stack. Send to Location works in the expected way.

**Comment by Cattlesquat** [comment]
Yes Player Hands have been a disaster as long as I can remember. Their code design/architecture doesn't lend itself to straightforward fixes, which is probably why no one has visited them in recent years.

**Comment by riverwanderer** [comment]
Player Hands are very useful though. To the editor/user, there doesn't seem to be much wrong with them and there was a good fix or two in v3.5.x.

For this issue there is a workaround, unfortunately one that requires additional complexity and knowledge. Whilst it persists, users may get confused and put off Player Hands should they stumble across it. I will put a note on my Player Hands "how to" thread if it can't be fixed.

**Comment by Cattlesquat** [comment]
I think what I usually do is have a button to "organize your hand" or whatever, and that uses some GKC to move every card in the hand off to somewhere, and then every card back on, and so then they all end up "right" again.

**Comment by riverwanderer** [comment]
@Cattlesquat  - In C&C Nap modules, I do it as a card enters or is moved in the hand; each card has a Trigger to check if it is on a row home location, if not it does a Send to Location to move itself to the home location of the row that it is in - then, as with your  "organize" button, Player Hand features kick-in and all is well. I developed this method without realising that one could do much the same just by using the features of an irregular grid, except one can't because this issue prevents that.

**Comment by riverwanderer** [comment]
Closing this as the bug did not occur during extensive play testing of a development module relying on the fix.
